### DISCUSSION TITLE
How are poetry commands calling my python interpreter?

### DISCUSSION THREAD

**Discussion by e-b** [python-poetry/poetry#9008] [Q&A] [answered]
When I execute "poetry install" or "poetry env info" i get [Errno 2] No such file or directory: 'python'
However, when I execute "poetry check" i get "All set!"
What am I missing?

I'm on OSX 14.2.1 on M1, Python 3.11.4, Poetry 1.7.1 and I set config virtualenvs.in-project = true

**Comment by abn** [comment]
Can you check if your virtual environment is created correctly?

```sh
.venv/bin/python --version
```

**Comment by e-b** [comment]
> Can you check if your virtual environment is created correctly?
> 
> ```shell
> .venv/bin/python --version
> ```

I removed it (because I though something might be broken) and now I cannot (re-)create it again.

**Reply by abn** [reply, marked as answer]
Try creating an environment manually perhaps?

```sh
python -m venv . venv
source . venv/bin/activate
poetry install
```

Another thing to maybe check is also if your poetry install is not broken. For example see if what interpreter poetry itself uses is available. 

You can see this using the following one liner. 

```sh
head -n5 $(which poetry)
```

If it is broken maybe a good idea to uninstall poetry and install it again. 

My personal preference is to use `pipx` to install poetry, you can also use the poetry install script.

**Reply by e-b** [reply]
Thank you, I ran this:
(venv) user@machine xyz % python -m venv . venv
(venv) user@machine xyz % poetry install
Updating dependencies
Resolving dependencies... (0.1s)

Writing lock file

Installing the current project: xyz (0.1.0)
The current project could not be installed: No file/folder found for package xyz
If you do not want to install the current project use --no-root
(venv) eriq@erics-air-2 xyz % 

I am in folder xyz but the tool tells me that "No file/folder found for package xyz"

will try uninstall and reinstall with pipx as well.

**Reply by abn** [reply]
By default poetry will look for a package named the same as your project name in the project root directory. 

 It would expect there to the a package defined manually. Especially if you didn't use `poetry new` or `poetry init`. 

In your case, in suspect you will will need at least an empty file at `xyz/xyz/__init__.py`.

**Reply by e-b** [reply]
excellent - thanks! I'll do that.

**Reply by abn** [reply]
Also note the prefix `(venv)`.

```
(venv) user@machine xyz %
```

Looks like the issue was you had a venv already activated and you might have deleted the already activated environment before deactivating it. JFYI.

**Reply by e-b** [reply]
yes, it was that - thanks!
I sorted it our.
